builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -762,30 +838,94 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	var mvm mo.VirtualMachine
 
 	collector := property.DefaultCollector(client.Client)
-	if err := collector.RetrieveOne(context.TODO(), vm.Reference(), []string{"guest", "summary", "datastore"}, &mvm); err != nil {
+	if err := collector.RetrieveOne(context.TODO(), vm.Reference(), []string{"guest", "summary", "datastore", "config"}, &mvm); err != nil {
 		return err
 	}
 
-	log.Printf("[DEBUG] %#v", dc)
-	log.Printf("[DEBUG] %#v", mvm.Summary.Config)
-	log.Printf("[DEBUG] %#v", mvm.Guest.Net)
+	log.Printf("[DEBUG] Datacenter - %#v", dc)
+	log.Printf("[DEBUG] mvm.Summary.Config - %#v", mvm.Summary.Config)
+	log.Printf("[DEBUG] mvm.Guest.Net - %#v", mvm.Guest.Net)
+
+	disks := make([]map[string]interface{}, 0)
+	for _, device := range mvm.Config.Hardware.Device {
+		if vd, ok := device.(*types.VirtualDisk); ok {
+
+			virtualDisk := vd.GetVirtualDevice()
+			disk := make(map[string]interface{})
+
+			log.Printf("[FINDME] virtualDisk info: %#v", virtualDisk)
+			log.Printf("[FINDME] virtualDisk backing info: %#v", virtualDisk.Backing.(*types.VirtualDiskFlatVer2BackingInfo))
+			log.Printf("[FINDME] virtualDisk thin?: %v", *virtualDisk.Backing.(*types.VirtualDiskFlatVer2BackingInfo).ThinProvisioned)
+			log.Printf("[FINDME] vd: %#v", vd)
+
+			thin := virtualDisk.Backing.(*types.VirtualDiskFlatVer2BackingInfo).ThinProvisioned
+			if thin != nil {
+				isThin := *thin
+				if isThin {
+					disk["type"] = "thin"
+				}
+				// TODO Investigate why disks created as either thin or eager_zeroed
+				// all return as `ThinProvisioned`: true and `EagerlyScrub`: nil
+				// } else {
+				// 	disk["type"] = "eager_zeroed"
+				// }
+			}
+
+			disk["key"] = virtualDisk.Key
+
+			diskPath := virtualDisk.Backing.(*types.VirtualDiskFlatVer2BackingInfo).FileName
+			diskPathSplit := strings.Split(diskPath, " ")
+			if len(diskPathSplit) == 2 {
```

Comment:
Do we want to document how we are doing this syntax?  Not sure if it is documented ... probably is